### ISSUE TITLE
[Semantics]: Not focus on element directly

### ISSUE THREAD

**Issue by guinardA** [flutter/flutter#97573]
My application is simple scaffold with a text widget. I want when I launch the application or open a new page, focus on a text or even on the title of my appBar and not the backButton for a new page.
On Android it does not focus directly, but only after a delayed.  And for iOS never focus either directly or with a delayed.

### Simple page with text widget focused after 1 second delayed

**Steps to Reproduce**
Using the following example app:
<details>
<summary>code sample 1</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(
    title: 'SimpleApp',
    home: Page(),
  ));
}

class Page extends StatefulWidget {
  const Page({
    Key? key,
  }) : super(key: key);

  @override
  _PageState createState() => _PageState();
}

class _PageState extends State<Page> {
  var display = false;

  @override
  void initState() {
    super.initState();

    WidgetsBinding.instance?.addPostFrameCallback((_) {
      Future.delayed(const Duration(seconds: 1)).then((onValue) => setState(() {
            display = true;
          }));
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('My page'),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          if (display == true) ...[
            Center(
              child: Semantics(
                focused: true,
                child: const Text('I want the focus !'),
              ),
            ),
          ],
        ],
      ),
    );
  }
}
```
</details>

1. Open the above app with talkback enabled on an Android device.
2. Wait 1 Second
3. The text "I want the focus" is appear and directly focused

If you change delayed time to 0 seconds, the text "I want the focus" appear directly and not focused. The focused stay on appBar title.

Not working on iOS, widget is never focused with the semantics focused to true.

### Page navigation, focused on appBar title from a new page

My primary objective is that each time a new page is opened, we focus on the title of the appBar and not on the backButton. But without delayed impossible to put the focus on the title of the appBar.

**Steps to Reproduce**
Using the following example app:

<details>
<summary>code sample 2</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(
    title: 'Navigation Basics',
    home: PageOne(),
  ));
}

class PageOne extends StatefulWidget {
  const PageOne({
    Key? key,
  }) : super(key: key);

  @override
  _PageOneSate createState() => _PageOneSate();
}

class _PageOneSate extends State<PageOne> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Page one'),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Center(
            child: ElevatedButton(
              child: const Text('Go Page two'),
              onPressed: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(builder: (context) => const PageTwo()),
                );
              },
            ),
          ),
        ],
      ),
    );
  }
}

class PageTwo extends StatefulWidget {
  const PageTwo({
    Key? key,
  }) : super(key: key);

  @override
  _PageTwoSate createState() => _PageTwoSate();
}

class _PageTwoSate extends State<PageTwo> {
  var display = false;

  @override
  void initState() {
    super.initState();

    WidgetsBinding.instance?.addPostFrameCallback((_) {
      Future.delayed(const Duration(seconds: 1)).then((onValue) => setState(() {
            display = true;
          }));
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Column(
          children: [
            if (display == true) ...[
              Semantics(
                focused: true,
                child: const Text('Page Two'),
              ),
            ],
          ],
        ),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: const [
          Center(
            child: Text('I want the focus !'),
          ),
        ],
      ),
    );
  }
}
```
</details>

1. Open the above app with talkback enabled on an Android device.
2. Go the page two
3. Wait 1 second
3. The appBar title "Page Two" is appear and directly focused

If you change delayed time to 0 seconds, the appBar title "Page Two" appear directly and not focused. The focused stay on appBar backButton.

Not working on iOS, widget is never focused with the semantics focused to true.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 11.6 20G165 darwin-x64, locale fr-FR)
    • Flutter version 2.8.1 at /usr/local/Caskroom/flutter/2.8.0/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/arnaud/Library/Android/sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (2 available)
    • Mi Note 10 (mobile) • 4182b5b • android-arm64  • Android 10 (API 29)
    • Chrome (web)        • chrome  • web-javascript • Google Chrome 97.0.4692.99

• No issues found!
```
</details>

**Comment by maheshmnj** [comment]
Hi @guinardA, Thanks for filing the issue. I ran your both code samples and I see the text is focused as intended with or without Duration (even with `Duration.zero`)

In code sample 1 The focus is shifted to center of the screen
In code sample 2 The focus is shifted to the appbar title

*Note: unmute the video to hear the talkback output*

Output 1 | Output 2
:-: | :-:
<video src='https://user-images.githubusercontent.com/31410839/151973295-f3ef4b2e-daf3-4628-a4a8-dbc25e4f81b8.mov' width=180/> | <video src='https://user-images.githubusercontent.com/31410839/151972869-713f5654-9167-4de7-8fa5-acc561556b05.mov' width=180/>

 I am a little confused is this issue on Android or IOS, the steps above mentions to reproduce on Android and mentions below that it doesn't work on IOS?

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-arm, locale en-GB)
    • Flutter version 2.8.1 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (3 available)
    • Redmi K20 Pro (mobile) • 192.168.1.2:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos            • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 97.0.4692.99

• No issues found!
```
</details>

**Comment by maheshmnj** [comment]
> For sample 2 in your video you have a delayed of 1 second. But with delayed of 0 second, the focus is not shifted to the appbar title on Android 10 and 12.

@guinardA This seems to be working as intended Please see this comment https://github.com/flutter/flutter/issues/97222#issuecomment-1022414938

cc: @chunhtai

**Comment by guinardA** [comment]
For me it's two different issues.

the #97222  is just on iOS and this one is for Android and iOS.

**Comment by chunhtai** [comment]
can be solved by https://github.com/flutter/flutter/issues/97747
